woocommerce/woocommerce-gateway-stripe
# [Deferred intent] Update the payment method billing details during checkout

Blocked by https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2816

Related to #2760 

## Changes proposed in this Pull Request:

- Introduce the WC_Stripe_Action_Scheduler_Service class, aimed to handle scheduling actions.
- When using a saved payment method in the checkout, update its billing details in Stripe async with the order billing details.

## Testing instructions

1. Ensure the new checkout experience is enabled
2. As a logged-in shopper, go to the My account > Payment methods > Add payment method
3. Add a new card, like 4242424242424242
4. Go to the Stripe dashboard > Customers > The customer associated with the shopper you used to add the card
5. Confirm the Payment method doesn't have any billing details associated with it
  <img width="700" alt="image" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/41606954/75be7210-1951-4d58-8c73-92dea8a67bca">

7. As the shopper, add a product to the cart and go to the checkout page
8. Fill up the checkout form to include the billing details
9. Select the card you just added and place the order
10. As the store admin, go to the WordPress dashboard > Scheduled Actions page ( `siteurl/wp-admin/tools.php?page=action-scheduler` )
11. Search for `wc_stripe_update_saved_payment_method`
12. Confirm there's an action with:
  - Status: Either "Pending" or "Complete". It could have run already
  - Arguments: 'payment_method' => The ID of the payment method we're updating, 'order_id' => The ID of the WC order you just placed
  - Group: "woocommerce_stripe"
  - Recurrence: "Non-repeating"
  - Scheduled date: Now or sometime in the past.

---


**Post merge**
